lib/logstash/outputs/scalyr.rb

```diff
@@ -661,6 +661,17 @@ def build_multi_event_request_array(logstash_events)
           ).force_encoding('UTF-8')
         end
         event_json = scalyr_event.to_json
+      rescue Java::JavaLang::ClassCastException => e
```

Comment:
Here we only convert it to JSON for internal "accounting" purposes aka to determine the total payload size.

We again convert the whole body the JSON later on (https://github.com/scalyr/logstash-output-scalyr/blob/5bba7f86f8172e97727eb9bc1334d9f9867804ce/lib/logstash/outputs/scalyr.rb#L763) so we need to handle the same issue there.

And yeah as discussed in the past,  serializing everything twice is not ideal, but manually concatenating partial JSON objects together is error prone and per micro-benchmarks and operational data serialization doesn't take that long and it's not worth the micro optimizations at this point.

Reply:
I think at this point all the events are already JSONified, we just handle them as strings as far as I can tell. Still it wouldn't hurt to catch the issue here again just in case, if it doesn't come up then no loss.

Reply:
Hm you sure?

https://github.com/scalyr/logstash-output-scalyr/blob/5bba7f86f8172e97727eb9bc1334d9f9867804ce/lib/logstash/outputs/scalyr.rb#L763, https://github.com/scalyr/logstash-output-scalyr/blob/5bba7f86f8172e97727eb9bc1334d9f9867804ce/lib/logstash/outputs/scalyr.rb#L735, https://github.com/scalyr/logstash-output-scalyr/blob/5bba7f86f8172e97727eb9bc1334d9f9867804ce/lib/logstash/outputs/scalyr.rb#L690

To me it looks like ``create_multi_event_request`` uses ``scalyr_events`` which is an array of hashes.

Or am I misreading it?

Reply:
Oh I think I misread it, these are hashes. But having said that they would be hashes that already ran into the issue earlier and got sanitized of Bignums, which is why I didn't run into the error here in the code in my manual testing.

Reply:
Ah, sorry, you are right - we sanitize the input hash and not the actual serialized string so it should be fine, yeah :)